types/object.go

```diff
@@ -5,6 +5,27 @@ import (
 	"sync/atomic"
 )
 
+// ObjectMode describes what users can operate on this object.
+//
+// Different object modes are orthogonal, and an object could have different
+// object modes at the same time.
+//
+// For example:
+//
+// - ModeDir means we can do list on this object('s path)
+// - ModeRead means we can read it as a normal file
+// - ModeLink means we can read this object's link target
+//
+// And we can compose them together:
+//
+// - ModeRead | ModeLink: Think about a symlink, we can still read it.
+// - ModeDir | ModeLink: Think about a symlink to dir.
+// - ModeAppend | ModeRead: file in fs could be both read and append.
+//
+// Reference
+//
+// - [GSP-25](https://github.com/beyondstorage/specs/blob/master/rfcs/25-object-mode.md)
+// - [Core Concept: Object](https://beyondstorage.io/docs/go-storage/internal/core-concept#object)
```

Comment:
btw, go doc is not rendered in markdown. Do we need something like a docstring style guide?

Reply:
Yes, we need a docstring style guide, maybe at the org level. I add an issue https://github.com/beyondstorage/site/issues/139 to track.